### PR TITLE
avoid fetchind data 2 times

### PR DESCRIPTION
**Pull request by lepetinez**

### CODE REVIEW

File: app/src/main/java/com/example/marcinwisniewski/intivetrainingapp/movielist/view/fragment/MovieListFragment.kt [hunk 1]
```diff
@@ -35,18 +34,24 @@ class MovieListFragment : Fragment() {
         movieListBinding.setLifecycleOwner(this)
         movieListViewModel = ViewModelProviders.of(this, movieListViewModelFactory).get(MovieListViewModel::class.java)
         movieListBinding.movieListViewModel = movieListViewModel
+        initList(movieListBinding)
```
Comment (awachowicz) [inline app/src/main/java/com/example/marcinwisniewski/intivetrainingapp/movielist/view/fragment/MovieListFragment.kt]: I think recyclerView as argument is enough, isn't it ?

**Review by awachowicz — APPROVED** [review]

File: app/src/main/java/com/example/marcinwisniewski/intivetrainingapp/moviedetails/view/fragment/MovieDetailsFragment.kt [hunk 2]
```diff
@@ -13,7 +13,7 @@ import com.example.marcinwisniewski.intivetrainingapp.moviedetails.viewmodel.Mov
 import com.example.marcinwisniewski.intivetrainingapp.movielist.model.Movie
 
 class MovieDetailsFragment : Fragment() {
-    //TODO: viewmodel by dependency injection
+    // TODO: viewmodel by dependency injection
```
Comment (Bartosz-Kozajda) [inline app/src/main/java/com/example/marcinwisniewski/intivetrainingapp/moviedetails/view/fragment/MovieDetailsFragment.kt]: i think this TODO is not needed anymore.

File: app/src/main/java/com/example/marcinwisniewski/intivetrainingapp/movielist/view/fragment/MovieListFragment.kt [hunk 3]
```diff
@@ -35,18 +34,24 @@ class MovieListFragment : Fragment() {
...
-        initAndObserveList(movieListBinding = getBinding<MovieListFragmentBinding>(view) as MovieListFragmentBinding)
+        if (savedInstanceState == null) {
+            fetchMovies()
+        }
     }
 
-    private fun initAndObserveList(movieListBinding: MovieListFragmentBinding) {
+    private fun initList(movieListBinding: MovieListFragmentBinding) {
         movieListBinding.root.recycler_list.layoutManager = LinearLayoutManager(context)
         val movieListAdapter = MovieListAdapter()
```
Comment (Bartosz-Kozajda) [inline app/src/main/java/com/example/marcinwisniewski/intivetrainingapp/movielist/view/fragment/MovieListFragment.kt:46]: DI

**Review by Bartosz-Kozajda — APPROVED** [review]

File: app/src/main/java/com/example/marcinwisniewski/intivetrainingapp/movielist/view/fragment/MovieListFragment.kt (hunk 3 above)
Comment (lepetinez) [reply app/src/main/java/com/example/marcinwisniewski/intivetrainingapp/movielist/view/fragment/MovieListFragment.kt:46]: will be added in next pull request

**Pull request merged**